authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -198,34 +194,237 @@ func getActionName(verb string) string {
+		authInfoList = append(authInfoList, authInfoSingle)
 	}
-	return authInfo
+	return authInfoList, nil
+}
+
+func getAuthInfoListForWildcard(subRevReq *authzv1.SubjectAccessReviewSpec, operationsMap azureutils.OperationsMap) ([]azureutils.AuthorizationActionInfo, error) {
+	var authInfoList []azureutils.AuthorizationActionInfo
+	var err error
+	if subRevReq.ResourceAttributes.Resource == "*" {
+		/*
+			This sections handles the following scenarios:
+
+			| Scenario              | Namespace is empty (Cluster scope call)                    | Namespace is not empty (NS scope)             |
+			------------------------ ------------------------------------------------------------  ----------------------------------------------
+			| Verb-*, Res-*, Group-*| All cluster and ns res with all verbs at clusterscope      | All ns resources at ns scope                  |
+
+			| Res-*, Group-*        | All cluster and ns res with specified verb at clusterscope | All ns res with specified verb at ns scope    |
+
+			| Verb-*, Res-*         | All cluster and ns res with all verbs under                | All ns res with all verbs under specified     |
+			|                       | specified apigroup at clusterscope                         | apigroup at nsscope                           |
+
+			| Resource - *          | All CS and NS Resources under specifed apigroup with       | All NS Resources under specifed apigroup with |
+			|                       | specified verb at cluster scope                            | specified verb at ns scope                    |
+		*/
+
+		filteredOperations := azureutils.NewOperationsMap()
+		if subRevReq.ResourceAttributes.Group == "*" {
+			// all resources under all apigroups
+			filteredOperations = operationsMap
+		} else if subRevReq.ResourceAttributes.Group != "" {
+			// all resources under specified apigroup
+			if value, found := operationsMap[subRevReq.ResourceAttributes.Group]; found {
+				filteredOperations[subRevReq.ResourceAttributes.Group] = value
+			} else {
+				return nil, errors.Errorf("No resources found for group %s", subRevReq.ResourceAttributes.Group)
+			}
+		} else {
+			// if Group is not there that means it is the core apigroup
+			if value, found := operationsMap["v1"]; found {
+				filteredOperations["v1"] = value
+			} else {
+				return nil, errors.New("No resources found for the core group")
+			}
+		}
+
+		// if Namespace is populated, we need to create the list only for namespace scoped resources
+		finalFilteredOperations := azureutils.NewOperationsMap()
+		if subRevReq.ResourceAttributes.Namespace != "" {
+			for group, resMap := range filteredOperations {
+				for resourceName, verbValues := range resMap {
+					for _, dataAction := range verbValues {
+						if dataAction.IsNamespacedResource {
+							finalFilteredOperations = initializeMapForGroupAndResource(finalFilteredOperations, group, resourceName)
+							finalFilteredOperations[group][resourceName] = verbValues
+							break
```

Comment:
Do we need to break here? Don't we need to add all data actions which are namespaced scope?

Reply:
discussed offline